lib/internal/readline/eventsToAsyncIteratorFactory.js

```diff
@@ -0,0 +1,114 @@
+'use strict';
+const {
+  Promise,
+  SymbolAsyncIterator,
+} = primordials;
+
+const PAUSE_THRESHOLD = 1024;
+const RESUME_THRESHOLD = 1;
+const ITEM_EVENTS = ['data'];
+const CLOSE_EVENTS = ['close', 'end'];
+const ERROR_EVENTS = ['error'];
+
+let FixedQueue;
+let setImmediate;
+function waitNext(emitter, next, events) {
+  return new Promise((resolve, reject) => {
+    const resolveNext = () => {
+      events.forEach((event) => emitter.off(event, resolveNext));
+      setImmediate(() => {
```

Comment:
I don't understand this change (why is there a setImmediate here?) and I am also not sure why we're doing explicit construction here (the `new Promise`) vs `timers/promises` event target.

Reply:
The promise is actually resolved in any of the events from the "events" array. I put a setImmediate here because I need that the call to next to be done after the listener responsible to add the new line to the queue.
Do you know if the calls to the event listeners are guaranteed to be done in order of inclusion? if that's the case, then I don't need the setImmediate here

Reply:
@benjamingr I'm looking into the EventEmitter code and it appears to be the case. I'll test the code without the setImmediate

Reply:
> Do you know if the calls to the event listeners are guaranteed to be done in order of inclusion? if that's the case, then I don't need the setImmediate here

There is an API to change it (via prependListener) but typically yes.